tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kvirtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("Infosource", func() {
```

Comment:
I think you should test all possible scenarios, which include:
- mac changed (the current covered scenario).
- missing interface in the guest.
- an additional interface in the guest.
- an interface which is in sync (domain + guest-agent).

This can still be a single test, but with multiple interfaces.

Reply:
bullet 2 and 3 are covered by the first bullet,
as once you change the MAC you get an interface which appear only in the domain (bullet 2),
and an interface that appears only on the guest (bullet 3)

I will add bullet 4 (planned to)

Reply:
I think that the outcome is the same, but not the scenario to begin with.
Several scenarios may reach the same outcome, but it makes sense to test them all to make sure we are not missing anything from a user perspective.

Reply:
added 2 + 3